app/controllers/transacs_controller.rb

```diff
@@ -0,0 +1,39 @@
+class TransacsController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @transacs = current_user.categories.find(params[:category_id]).transacs
+  end
+
+  def show
+    @transacs = current_user.categories.find(params[:category_id]).transacs.find(params[:id])
+  end
+
+  def new
+    @categories = current_user.categories
+    @transac = current_user.categories.find(params[:category_id]).transacs.new
+  end
+
+  def create
+    @categories = current_user.categories
+    @transac = @categories.find(params[:category_id]).transacs.create(transac_params)
+    @transac.save
+    redirect_to user_category_path(current_user, params[:category_id])
+    # if @transac.save
+    #   redirect_to user_category_path(current_user, )
+    # else
+    #   render :new
+    # end
```

Comment:
- [ ] Here as well. I would recommend removing the commented code to make the codebase cleaner.